sycl/test-e2e/bindless_images/array/fetch_handle_carray2d.cpp

```diff
@@ -0,0 +1,98 @@
+// REQUIRES: aspect-ext_oneapi_bindless_images
+// XFAIL: level_zero
+// UNSUPPORTED: hip
+// UNSUPPORTED-INTENDED: Undetermined issue in 'create_image' in this test.
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} env NEOReadDebugKeys=1 UseBindlessMode=1 UseExternalAllocatorForSshAndDsh=1 %t.out
+
+#include <sycl/sycl.hpp>
+
+int main() {
+
+  sycl::queue q{};
+
+  // Initialize input data
+  constexpr size_t width = 512;
+  std::vector<float> dataIn(width);
+  std::vector<float> dataOut(width);
+  for (int i = 0; i < width; i++) {
+    dataIn[i] = static_cast<float>(i);
+  }
+
+  // Image descriptor - can use the same for both images
+  sycl::ext::oneapi::experimental::image_descriptor desc(
+      sycl::range{width}, 1, sycl::image_channel_type::fp32);
+
+  // Extension: returns the device pointer to the allocated memory
+  sycl::ext::oneapi::experimental::image_mem imgMemoryIn(desc, q);
+  sycl::ext::oneapi::experimental::image_mem imgMemoryOut(desc, q);
+
+  // Extension: create the image and return the handle
+  sycl::ext::oneapi::experimental::unsampled_image_handle imgIn =
+      sycl::ext::oneapi::experimental::create_image(imgMemoryIn, desc, q);
+  sycl::ext::oneapi::experimental::unsampled_image_handle imgOut =
+      sycl::ext::oneapi::experimental::create_image(imgMemoryOut, desc, q);
+
+  void *imageHandlePtrGen = static_cast<void *>(sycl::malloc_device(
+      sizeof(sycl::ext::oneapi::experimental::unsampled_image_handle), q));
+  q.memcpy(static_cast<void *>(imageHandlePtrGen),
```

Comment:
I think it would be good to add a short comment to each copy about what it's supposed to be doing. In this case we're copying the image handle from the host to an image handle on device. But it gets more complicated with the copies below.

Reply:
Done, thanks